backend/middlewares/errorMiddleware.ts

```diff
@@ -0,0 +1,21 @@
+import { Request, Response, NextFunction } from 'express';
+import { Prisma } from '@prisma/client';
+
+const handleError = async (err: Prisma.PrismaClientKnownRequestError | string, req: Request, res: Response, next: NextFunction) => {
+
+    if (typeof (err) === 'string') return res.status(401).json(err);
+
+    let responseMsg;
+    switch (err.code) {
+        case 'P2002':
+        case 'P2025':
+            responseMsg = err.message.split('\n')[8].trim();
+            break;
+        default:
+            responseMsg = "Something Went Wrong!"
+            break;
+    }
+    res.status(401).json(responseMsg);
```

Comment:
We are sending a 401 for all of the errors, I think the error codes should be related to the actual error.

Reply:
This is handled in the latest commit of the branch.